src/fileio.h

```diff
@@ -22,41 +24,15 @@ struct LoadingError : public std::exception {
 	std::string message;
 };
 
-/**
- * Base class for reading the contents of a directory.
- * Intended use:
- * - After creation (with the #MakeDirectoryReader function), call #OpenPath to open the directory.
- * - A new entry of the directory is returned with every call to #NextEntry.
- * - At the end, call #ClosePath to allow the object to clean up, and get ready for a next use.
- *
- * Alternatively, use #MakePath to construct a path you want to examine.
- * @ingroup fileio_group
- */
-class DirectoryReader {
-public:
-	DirectoryReader() = default;
-	virtual ~DirectoryReader() = default;
-
-	virtual void OpenPath(const char *path) = 0;
-	virtual const char *NextEntry() = 0;
-	virtual void ClosePath() = 0;
-
-	const char *NextFile();
-	virtual const char *MakePath(const char *directory, const char *fname) = 0;
-
-	virtual bool EntryIsFile() = 0;
-	virtual bool EntryIsDirectory() = 0;
-};
-
-extern const char *DIR_SEP;  ///< Directory separator character.
+extern const char DIR_SEP;  ///< Directory separator character.
```

Comment:
This won't work if we change the definition to use `auto`. How about just moving the definition from the source file to the header?

Reply:
For some reason the compiler decided that std::filesystem is undeclared there, lol

Reply:
Okay, I figured this out, for some reason rcdgen used C++ 11 still, I updated the CXX_STANDARD for that